src/Tribe/Views/V2/Views/Traits/iCal_Data.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Provides methods for getting iCal data for views
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views
+ */
+
+namespace Tribe\Events\Views\V2\Views\Traits;
+
+use Tribe\Events\Views\V2\View;
+
+/**
+ * Trait iCal_Data
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views\Traits
+ *
+ * @property string $string The slug of the View instance.
+ */
+trait iCal_Data {
+	/**
+	 * Returns the iCal data we're sending to the view.
+	 *
+	 * @since TBD
+	 *
+	 * @return object
+	 */
+	protected function get_ical_data() {
+		$slug = $this->slug;
+
+		/**
+		 * A filter to control whether the "iCal Import" link shows up or not.
+		 *
+		 * @since unknown
+		 *
+		 * @param boolean $show Whether to show the "iCal Import" link; defaults to true.
+		 */
+		$display_ical = apply_filters( 'tribe_events_list_show_ical_link', true );
+
+		/**
+		 * Allow for customization of the iCal export link "Export Events" text.
+		 *
+		 * @since unknown
+		 *
+		 * @param string $text The default link text, which is "Export Events".
+		 */
+		$link_text = apply_filters(
+			'tribe_events_ical_export_text',
+			sprintf(
+				/* translators: %s: Events (plural). */
+				__( 'Export %s', 'the-events-calendar' ),
+				tribe_get_event_label_plural()
+			)
+		);
+
+		$link_title = __( 'Use this to share calendar data with Google Calendar, Apple iCal and other compatible apps', 'the-events-calendar' );
+
+		switch( $slug ) {
+			case 'month':
```

Comment:
* :no_entry_sign: Space after opening control structure is required
* :no_entry_sign: No space before opening parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after SWITCH keyword; 0 found
* :no_entry_sign: There must be a space between switch and the first parenthesis '('

<sub>hash: c3fd644b3ed0f5bf476d5f9c88866795</sub>